### PR TITLE
[Console-IoT]- Add ID for IoT certificates

### PR DESCRIPTION
**Pull request by suyash-naithani**
### Type of change

- Refactoring

### Description

 Add meaningful ID for IoT certificates

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: console/console-init/ui/src/modules/iot-certificates/components/IoTCertificate/IoTCertificate.tsx [hunk 1]
```diff
@@ -37,7 +37,10 @@ export const IoTCertificate: React.FunctionComponent<IIoTCertificateProps> = ({
   const [onEditMode, setOnEditMode] = useState<boolean>(false);
 
   return (
-    <PageSection className={css(style.no_bottom_padding)}>
+    <PageSection
```
Comment (drcoolsanjeev) [inline console/console-init/ui/src/modules/iot-certificates/components/IoTCertificate/IoTCertificate.tsx:40]: page section doesn't need any id

File: console/console-init/ui/src/modules/iot-certificates/components/CertificateCard/CertificateCard.tsx [hunk 2]
```diff
@@ -62,17 +62,17 @@ export const CertificateCard: React.FunctionComponent<ICertificateCardProps> = (
 
   const dropdownItems = [
     <DropdownItem
-      id={`cc-dropdown-edit-${id}`}
+      id={`cert-card-edit-dropdown-${id}`}
```
Comment (ajaypratap003) [inline console/console-init/ui/src/modules/iot-certificates/components/CertificateCard/CertificateCard.tsx:65]: replace dropdown with dropdownitem.
 id={`cert-card-edit-dropdownitem-${id}`}

File: console/console-init/ui/src/modules/iot-certificates/components/CertificateCard/CertificateCard.tsx [hunk 3]
```diff
@@ -62,17 +62,17 @@ export const CertificateCard: React.FunctionComponent<ICertificateCardProps> = (
...
       key="edit"
       aria-label="Edit certificate"
       onClick={onEditCertificate}
     >
       Edit
     </DropdownItem>,
     <DropdownItem
-      id={`cc-dropdown-delete-${id}`}
+      id={`cert-card-delete-dropdown-${id}`}
```
Comment (ajaypratap003) [inline console/console-init/ui/src/modules/iot-certificates/components/CertificateCard/CertificateCard.tsx:73]: id={`cert-card-delete-dropdownitem-${id}`}

File: console/console-init/ui/src/modules/iot-certificates/components/IoTCertificate/IoTCertificate.tsx (hunk 1 above)
Comment (suyash-naithani) [reply console/console-init/ui/src/modules/iot-certificates/components/IoTCertificate/IoTCertificate.tsx:40]: I will let it be in this one but will keep in mind in other modules

**Comment by suyash-naithani** [comment]
Resolved all comments Looks good to merge

File: console/console-init/ui/src/modules/iot-certificates/components/CertificateForm/CertificateForm.tsx [hunk 4]
```diff
@@ -162,12 +164,12 @@ export const CertificateForm: React.FunctionComponent<ICertificateFormProps> = (
...
             <GridItem span={6}>
               <FormGroup
-                fieldId={`cf-not-before-${id}`}
+                fieldId={`cert-form-not-before-text-${id}`}
                 isRequired
                 label={getLabelByKey("not-before")}
               >
                 <TextInput
-                  id={`cf-not-before-${id}`}
+                  id={`cert-form-not-before-text-${id}`}
```
Comment (anukritijha) [inline console/console-init/ui/src/modules/iot-certificates/components/CertificateForm/CertificateForm.tsx:172]: As stated earlier, this is an input field, and the id doesn't signify that. Either the id should have 'text-box' or 'input'.

File: console/console-init/ui/src/modules/iot-certificates/components/CertificateCard/CertificateCard.tsx [hunk 5]
```diff
@@ -142,15 +151,23 @@ export const CertificateCard: React.FunctionComponent<ICertificateCardProps> = (
...
-            <Title headingLevel="h1" size="md" id={`cc-not-before-title-${id}`}>
+            <Title
+              headingLevel="h1"
+              size="md"
+              id={`cert-card-not-before-title-${id}`}
+            >
               <b>{getLabelByKey("not-before")}</b>
             </Title>
           </GridItem>
           <GridItem span={10} className={rowMargin}>
```
[the remark this answers is not in the document]
Comment (suyash-naithani) [reply console/console-init/ui/src/modules/iot-certificates/components/CertificateCard/CertificateCard.tsx:162]: ID left out on grid items

File: console/console-init/ui/src/modules/iot-certificates/components/CertificateForm/CertificateForm.tsx [hunk 6]
```diff
@@ -115,12 +115,12 @@ export const CertificateForm: React.FunctionComponent<ICertificateFormProps> = (
...
         <Form>
           <FormGroup
-            fieldId={`cf-subjectdn-${id}`}
+            fieldId={`cert-form-subjectdn-${id}`}
             isRequired
             label={getLabelByKey("subject-dn")}
           >
             <TextInput
-              id={`cf-subjectdn-${id}`}
+              id={`cert-form-subjectdn-${id}`}
```
[the remark this answers is not in the document]
Comment (suyash-naithani) [reply console/console-init/ui/src/modules/iot-certificates/components/CertificateForm/CertificateForm.tsx]: You're right. Made changes :)

File: console/console-init/ui/src/modules/iot-certificates/components/CertificateCard/CertificateCard.tsx [hunk 7]
```diff
@@ -82,28 +82,36 @@ export const CertificateCard: React.FunctionComponent<ICertificateCardProps> = (
...
-            <Title headingLevel="h1" size="md" id={`cc-subject-dn-title-${id}`}>
+            <Title
+              headingLevel="h1"
+              size="md"
+              id={`cert-card-subject-title-${id}`}
+            >
               <b>{getLabelByKey("subject-dn")}</b>
             </Title>
           </GridItem>
           <GridItem span={10} className={rowMargin}>
```
[the remark this answers is not in the document]
Comment (anukritijha) [reply console/console-init/ui/src/modules/iot-certificates/components/CertificateCard/CertificateCard.tsx:106]: I believe this displays the value of 'subject-dn', which is why I think an id would be needed.

File: console/console-init/ui/src/modules/iot-certificates/components/CertificateCard/CertificateCard.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (anukritijha) [reply console/console-init/ui/src/modules/iot-certificates/components/CertificateCard/CertificateCard.tsx:162]: As stated above, I believe an id would be required here.

**Pull request merged**